extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java

```diff
@@ -199,6 +199,13 @@
     @ConfigItem
     public Optional<String> multitenantSchemaDatasource;
 
+    /**
+     * This setting is used to control whether we should consult the JDBC metadata to determine certain Settings default values.
+     * In some specific cases the detection is not necessary and speeds up the application startup.
```

Comment:
This is very true.

Which make me wonder if we could evaluate such things only as necessary: lazily triggered by demand.

That would be nicer as:

- everyone would benefit from the same benefits of this flag when it's a correct optimisation
- people don't need to configure things
- when it is absolutely incorrect to set this, the optimisation would automatically get disabled

Reply:
I don't know if this helps, but I noticed that the slowness in my case is due to loading of UDTs: https://www.postgresql.org/docs/9.5/xtypes.html